src/Http/Routing/src/RouteOptions.cs

```diff
@@ -89,6 +90,16 @@ internal ICollection<EndpointDataSource> EndpointDataSources
     /// </summary>
     internal IDictionary<string, Type> TrimmerSafeConstraintMap => _constraintTypeMap;
 
+    /// <summary>
+    /// Add Regex-based constraints to the constraint map (e.g. alpha and regex constraint types). This is called automatically by <see cref="RoutingServiceCollectionExtensions.AddRouting(IServiceCollection)"/>.
+    /// </summary>
+    public void AddRegexConstraints()
+    {
+        // Regex-based constraints
+        AddConstraint<AlphaRouteConstraint>(_constraintTypeMap, "alpha");
```

Comment:
I think we would only do that if we were committing to implement a non-regex based alpha constraint. Otherwise there isn't much point in this API at all (no size benefit).

Reply:
... I guess we could use the source generator for the alpha constraint.

Reply:
> ... I guess we could use the source generator for the alpha constraint.

We already do. See https://github.com/dotnet/aspnetcore/issues/46142#issuecomment-1405223258